src/main/java/org/tron/program/Benchmark.java

```diff
@@ -0,0 +1,378 @@
+    long physicalMemorySize = os.getTotalPhysicalMemorySize();
+    System.out.println("old way memory: "+ physicalMemorySize / 1024 / 1024 +" MB");
+
+    MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+    Object attribute = null;
+    try {
+      attribute = mBeanServer.getAttribute(new ObjectName("java.lang","type","OperatingSystem"), "TotalPhysicalMemorySize");
+      long i = (Long) attribute;
+      System.out.println("new way memory: "+ i / 1024 / 1024 +" MB");
+    } catch (MBeanException e) {
+      e.printStackTrace();
+    } catch (AttributeNotFoundException e) {
+      e.printStackTrace();
+    } catch (InstanceNotFoundException e) {
+      e.printStackTrace();
+    } catch (ReflectionException e) {
+      e.printStackTrace();
+    } catch (MalformedObjectNameException e) {
+      e.printStackTrace();
+    }
+
+    return (long)(physicalMemorySize * 1.0 / 1024 / 1024);
+  }
+
+  public boolean checkJavaVersion() {
+    String name = System.getProperty("java.runtime.name").toLowerCase();
+    if (name.indexOf("openjdk") != -1) {
+      return false;
+    }
+    String[] version = System.getProperty("java.vm.specification.version").split("\\.");
+    if (version.length < 2) {
+      return false;
+    } else if (Integer.valueOf(version[0]) * 10000 + Integer.valueOf(version[1]) < 10008) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  public int checkCPUCore() {
+    return Runtime.getRuntime().availableProcessors();
+  }
+
+
+  public String readFile(String fileName) {
+
+    try (BufferedReader br = new BufferedReader(new InputStreamReader(Benchmark.class.getResourceAsStream(fileName)))) {
+      return br.lines().collect(Collectors.joining(System.lineSeparator()));
+    } catch(IOException e) {
+      logger.info(e.getMessage());
+      return null;
+    }
+
+  }
+
+  public void writeFile(String content, String fileName) {
+
+    File file = new File(fileName);
+    try (Writer writer = new FileWriter(file)){
+      writer.write(content);
+    } catch (IOException e) {
+      logger.info(e.getMessage());
+    }
+  }
+
+  public static void main(String[] args) {
+
+    Benchmark benchmark = new Benchmark();
+    boolean checkVersion = benchmark.checkJavaVersion();
+    int systemExit = 0;
+
+    if (checkVersion) {
+      System.out.println("1. JAVA VERSION:\nsatisfied");
+    } else {
+      System.out.println("1. JAVA VERSION:\nbefore run java-tron, java MUST be oracle jdk, and version >= 1.8");
+      systemExit = 1;
+    }
+
+    int cpuCore = benchmark.checkCPUCore();
+    int minCpuCore = 16;
+    if (cpuCore >= minCpuCore) {
+      System.out.println("2. CPU CORE:\nsatisfied");
+    } else {
+      System.out.println("2. CPU CORE:\ndo not update new verson java-tron, because of too few CPU cores, "
+          + "it needs at least " + minCpuCore + " cores");
+      systemExit = 1;
+    }
+
+    long minMem = 30 * 1024; // 30G
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Cast one of the operands of this multiplication operation to a "long". ([squid:S2184](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2184))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1546&issues=AWXxSwHSxJR_uy9BGF1H&open=AWXxSwHSxJR_uy9BGF1H)

[ISSUE_KEY]: AWXxSwHSxJR_uy9BGF1H